### PR TITLE
Add some recommendations in Learn guides

### PR DESCRIPTION
**Pull request by tomymehdi**
Hello OpenZeppelin team, hope you are doing great

Thank you for such a great documentation. After doing all the steps both in truffle and hardhat I detected some possible enhances and just jumped and created this fork to do the pull request.

Hope it's usefull,
Best Regards,

### CODE REVIEW

File: components/learn/modules/ROOT/pages/setting-up-a-node-project.adoc [hunk 1]
```diff
@@ -16,6 +16,19 @@ TIP: If you are running Windows consider using https://docs.microsoft.com/en-us/
...
+--
+
+[.hardhat]
+--
+Current Hardhat node version recommendation is using LTS. 
+https://hardhat.org/tutorial/setting-up-the-environment[Hardhat node version]
+https://nodejs.org/en/about/releases/[Current LTS version]
+--
+
+
```
Comment (frangio) [inline components/learn/modules/ROOT/pages/setting-up-a-node-project.adoc]:
```suggestion
Once you're done, run `node --version` on a terminal to check your installation. We recommend using the https://nodejs.org/en/about/releases/[active LTS version] for best compatibility with Ethereum libraries.
```

File: components/learn/modules/ROOT/pages/upgrading-smart-contracts.adoc [hunk 2]
```diff
@@ -487,6 +487,24 @@ To test upgradeable contracts we should create unit tests for the implementation
 
 When we want to upgrade, we should create unit tests for the new implementation contract, along with creating higher level tests for testing interaction via the proxy after we upgrade using `upgradeProxy`, checking that state is maintained across upgrades.
 
+== Possible issues
+
+While learning how to upgrade contract you might find yourself in a situation of conflicting contract on the local environment.
```
Comment (frangio) [inline components/learn/modules/ROOT/pages/upgrading-smart-contracts.adoc]: What do you mean by "conflicting contract"?

File: components/learn/modules/ROOT/pages/writing-automated-tests.adoc [hunk 3]
```diff
@@ -34,6 +34,17 @@ We'll use https://www.chaijs.com/[Chai] assertions for our unit tests.
 $ npm install --save-dev chai
 ```
 
+[.hardhat]
+--
+Add @nomiclabs/hardhat-truffle5 in hardhat.config.js
```
Comment (frangio) [inline components/learn/modules/ROOT/pages/writing-automated-tests.adoc]:
```suggestion
For compatibility with OpenZeppelin Test Helpers with Harhdat, dd `@nomiclabs/hardhat-truffle5` in hardhat.config.js. We recommend using https://hardhat.org/hardhat-chai-matchers/docs/overview[Hardhat Chai Matchers] as a better supported alternative for Ethers.js.
```

**Review by frangio — COMMENTED** [review]
Thank you @tomymehdi!

File: components/learn/modules/ROOT/pages/upgrading-smart-contracts.adoc (hunk 2 above)
Comment (tomymehdi) [reply components/learn/modules/ROOT/pages/upgrading-smart-contracts.adoc]: When all deploy scripts are run if the `npx hardhat node` is not restarted it wont allow to deploy all contracts

Comment (tomymehdi) [reply components/learn/modules/ROOT/pages/upgrading-smart-contracts.adoc]: @frangio also in some cases `npx hardhat clean` needed to be run

**Comment by tomymehdi** [comment]
@frangio just rebased

Kind regards

**Review by frangio — APPROVED** [review]
Thank you

**Pull request merged**